src/app/data-planes/data/index.ts

```diff
@@ -208,20 +200,68 @@ export const SidecarDataplane = {
     }
   },
 }
-
-export const InspectRules = {
-  fromCollection(partialInspectRules: PartialInspectRulesForDataplane): InspectRulesForDataplane & {} {
-    const rules = Array.isArray(partialInspectRules.rules) ? partialInspectRules.rules : []
-    const proxyRules = getProxyRules(rules)
-    const toRules = getToRules(rules)
-    const fromRuleInbounds = getFromRuleInbounds(rules)
-
+export type Rule = Omit<InspectBaseRule, 'conf' | 'origin'> & {
+  type: string
+  ruleType: string
+  inbound?: InspectInbound
+  config: InspectBaseRule['conf']
+  origins: InspectBaseRule['origin']
+}
+export type RuleCollection = Omit<PartialInspectRulesForDataplane, 'rules'> & {
+  // httpMatches: unknown[]
+  rules: Rule[]
+}
+export const Rule = {
+  fromObject(item: InspectBaseRule): Rule {
+    const { conf, origin, matchers, ...rest } = item
+    return {
+      type: '',
+      ruleType: '',
+      ...rest,
+      config: Object.keys(conf || {}).length > 0 ? conf : {},
+      origins: Array.isArray(origin) ? origin : [],
+      matchers: Array.isArray(matchers) ? matchers : [],
+    }
+  },
+  fromCollection(partialInspectRules: PartialInspectRulesForDataplane): RuleCollection {
```

Comment:
These are just functions that you would usually inline into `.map`, `.reduce` etc.

I've disabled eslint multiline ternaries in this file, which makes things a lot prettier instead of the automatic uglifying that that rule applies. lmk

Reply:
> I've disabled eslint multiline ternaries in this file, which makes things a lot prettier instead of the automatic uglifying that that rule applies. lmk

Besides what we already discussed, this changes makes this effect even worse because it makes ternaries harder to read, **especially** if they surround large expressions. It’s much better to wrap longer ternary expressions such that the `?` and `:` symbols start off on a new line as you can then see that there even is a ternary at play.

And just maybe the fact that things start looking ugly with this amount of inline expressions would be an indicator for when things should separate functions.

Reply:
Here's the link that @lahabana suggested reading in our sync https://copyconstruct.medium.com/small-functions-considered-harmful-91035d316c29 .

We seem to hit the issues outlined in this article many many times (over eager "abstractions", DRYing up code to the fullest extent possible, extremely verbose naming etc etc). There's a good quote here in relation to the small functions:

> I’ve worked on codebases inherited from folks who’d internalized this idea to such an unholy extent that the end result was pretty hellish and entirely antithetical to all the good intentions the road to it was paved with. In this post, I hope to explain why some of the oft-touted benefits don’t always pan out the way one hopes and the times when some of the ideas can actually prove to be counterproductive.

I really really want to avoid anything that takes this codebase back to where it used to be, and this includes avoiding the things mentioned above - including these "small functions" that offer no benefit.

What did you think of some of the arguments presented in the article? Do you there there are any takeaways for the situation here? Do you respect any of the authors points?

---

> It’s much better to wrap longer ternary expressions such that the ? and : symbols start off on a new line as you can then see that there even is a ternary at play.

Well, that again is a matter of opinion, I prefer it how it is now. I'd be happy to put it back to how it was before I turned off this rule if you like, I thought it would help 🤷 , lmk.

Reply:
>> We seem to hit the issues outlined in this article many many times (over eager "abstractions", DRYing up code to the fullest extent possible, extremely verbose naming etc etc).

> They have nothing to do with eager abstractions or "DRYing up code to the fullest extent possible" or "extremely verbose naming etc etc" (the longest word variable name were getFromRuleInbounds and ruleEntriesByInbound).

I think I might have not made it clear enough that these are things we quite often discuss in PRs generally, I mentioned them as I think the linked article is overall an extremely good article that group all these things under the title of "Small functions considered harmful"

Leaving our PR aside for a moment

> What did you think of some of the arguments presented in the article? Do you respect any of the authors points?

Reply:
> Please, do so.

I just added a commit to undo the ternary format amend.

Let me know if that means we can resolve this one.